apps/storybook/components/funke/IdCardPinScreen.stories.tsx

```diff
@@ -0,0 +1,243 @@
+}
+
+const stateMapping = {
+  enterPin: { title: 'Enter passport pin', icon: 'locked' },
+  loading: { title: 'Setting up identity', icon: 'loading' },
+  complete: { title: 'You wallet is ready', icon: 'complete' },
+} as const
+
+const iconMapping = {
+  locked: <HeroIcons.LockClosed color="$white" />,
+  loading: <HeroIcons.ArrowPath color="$white" />,
+  complete: <HeroIcons.ShieldCheck color="$white" />,
+} as const
+
+interface IdCardProps {
+  icon: keyof typeof iconMapping
+  issuerImage: string
+  userName?: string
+}
+
+function IdCard({ icon, issuerImage, userName }: IdCardProps) {
+  return (
+    <YStack gap="$6" p="$5" borderRadius="$8" overflow="hidden" borderWidth={1} borderColor="rgba(216, 218, 200, 1)">
+      <LinearGradient
+        colors={['#EFE7DA', '#EDEEE6', '#E9EDEE', '#D4D6C0']}
+        start={[0.98, 0.02]}
+        end={[0.28, 1.0]}
+        locations={[0.0207, 0.3341, 0.5887, 1.0]}
+        style={StyleSheet.absoluteFillObject}
+      />
+      <LinearGradient
+        colors={['rgba(255, 255, 255, 0.2)', 'rgba(255, 255, 255, 0.2)']}
+        start={[0, 0]}
+        end={[1, 0]}
+        style={StyleSheet.absoluteFillObject}
+      />
+      <XStack justifyContent="space-between">
+        <YStack gap="$1">
+          <Paragraph secondary>Personalausweis</Paragraph>
+          <Paragraph size="$6" fontWeight="$regular">
+            {userName ?? '********'}
+          </Paragraph>
+        </YStack>
+        <Stack>
+          <Image src={issuerImage} width={75} height={75} resizeMode="contain" />
+        </Stack>
+      </XStack>
+      <XStack justifyContent="flex-end">
+        {/* TODO: background color variables */}
+        <Circle size="$3" backgroundColor={icon === 'complete' ? '#C9D6BD' : '#02010033'}>
+          {iconMapping[icon]}
+        </Circle>
+      </XStack>
+    </YStack>
+  )
+}
+
+const IdCardPinScreen = ({ pinLength, onPinComplete, state, onGoToWallet, userName }: IdCardPinScreenProps) => {
+  const { title, icon } = stateMapping[state]
+  const [pin, setPin] = useState<string[]>(new Array(pinLength).fill(''))
+  const inputRefs = useRef<Array<TextInput | null>>([])
+  const programmaticFocus = useRef(false)
+
+  const isIndexFocusable = (index: number) => {
+    const firstFocusableIndex = pin.findIndex((digit) => digit === '')
+    return firstFocusableIndex === -1 || index <= firstFocusableIndex
+  }
+
+  const setPinValue = (digit: string, index: number) => {
+    const sanitized = digit.replace(/[^0-9]/g, '')
+    if (sanitized === '') return
+
+    setPin((pin) => {
+      const newPin = [...pin]
+      newPin[index] = sanitized
+      return newPin
+    })
+
+    // Focus next input
+    if (index < pin.length - 1) {
+      programmaticFocus.current = true
+      inputRefs.current[index + 1]?.focus()
+    } else if (index === pin.length - 1) {
+      inputRefs.current[index]?.blur()
+      onPinComplete(pin.join(''))
+    }
+  }
+
+  const focusInput = (index: number) => {
```

Comment:
Could we not make it so this can never be unfocussed?

Reply:
I'll think about it

Reply:
I looked into it, and it's quite tricky with the all the input components, and also having to handle e.g. starting over with the pin.